compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmPackageGen.java

```diff
@@ -545,13 +543,10 @@ private boolean listenerDeclarationFound(List<BIRGlobalVariableDcl> variableDcls
      *
      * @param module           bir module
      * @param initClass        module init class name
-     * @param interopValidator interop validator instance
      * @param isEntry          is entry module flag
      * @return The map of javaClass records on given source file name
      */
-    private Map<String, JavaClass> generateClassNameLinking(BIRPackage module, String initClass,
-                                                            InteropValidator interopValidator,
-                                                            boolean isEntry) {
+    private Map<String, JavaClass> generateClassNameLinking(BIRPackage module, String initClass, boolean isEntry) {
```

Comment:
Any way we could check and see whether we can get rid of this flag `isEntry` and have separate functions.? That given better readability IMO.

Reply:
isEntry is passed from generate module to each function generation. This is only used for add default Boolean params to function signature. So code will be duplicate and much complex code if try to separate them to functions. IMO correct thing is to move all desugar code (including encodes, record inits) from codegen to another class which needs to be done separately.